### ISSUE TITLE
How to instrument tomcat's CoyoteAdapter.service()?

### ISSUE THREAD

**Issue by RonaldQin** [jboss-javassist/javassist#283]
I want to instrument tomcat's CoyoteAdapter.service(). To let application execute instrumented code when user request. Below is my transformer's code.

       `@Override
	public byte[] transform(ClassLoader loader, String className, Class<?> classBeingRedefined,
			ProtectionDomain protectionDomain, byte[] classfileBuffer) throws IllegalClassFormatException {
		byte[] byteCode = classfileBuffer;
		String finalTargetClassName = this.targetClassName.replaceAll("\\.", "/");
		if (!className.equals(finalTargetClassName)) {
			return byteCode;
		}
		if (className.equals(finalTargetClassName) && loader.equals(targetClassLoader)) {
			System.out.println("[Instrument Agent] Transforming class " + className + " -- " + loader.toString());
			try {
				ClassPool pool = new ClassPool();
				pool.appendSystemPath();
				pool.appendClassPath(".:/home/lace/Documents/workspace-sts-3.9.10.RELEASE/Tomcat7/bin");  // Tomcat source code

				CtClass cc = pool.get(targetClassName);

				CtMethod m = cc.getDeclaredMethod(METHOD_NAME);
				System.out.println("[Instrument Agent] ----> " + m.getSignature() + ":" + m.getMethodInfo());
				m.insertBefore("System.out.println(\"[Instrumentation Agent] Agent information.\");");
				byteCode = cc.toBytecode();
				cc.detach();
			} catch (Exception e) {
				e.printStackTrace();
			}
		}
		return byteCode;
	}`

But, using above code, it can not output '[Instrumentation Agent] Agent information.' when I request Tomcat web applications. Anyone know why?
